core/src/main/java/io/hyperfoil/core/generators/BaseValueProviderBuilder.java

```diff
@@ -0,0 +1,52 @@
+package io.hyperfoil.core.generators;
+
+import io.hyperfoil.api.config.InitFromParam;
+
+public abstract class BaseValueProviderBuilder<P, V> implements InitFromParam<BaseValueProviderBuilder<P, V>> {
+   protected final P parent;
+   protected final V defaultValue;
+   protected V value;
+   protected String fromVar;
+
+   public BaseValueProviderBuilder(P parent, V defaultValue) {
+      this.parent = parent;
+      this.defaultValue = defaultValue;
+   }
+
+   public P end() {
+      return parent;
+   }
+
+   /**
+    * Initialize with a constant value.
+    *
+    * @param param Constant value.
+    * @return Self.
+    */
+   @Override
+   public abstract BaseValueProviderBuilder<P, V> init(String param);
+
+   /**
+    * Initialize with a constant value.
+    *
+    * @param value Constant value.
+    * @return Self.
+    */
+   public BaseValueProviderBuilder<P, V> value(V value) {
```

Comment:
In order to support fluent API it would be better to return the actual self-type - see e.g. https://github.com/Hyperfoil/Hyperfoil/blob/3001e8ece3c82b06b44c08d582ced356a62e81ea/core/src/main/java/io/hyperfoil/core/builders/IntConditionBuilder.java#L38

Reply:
Removed `BaseValueProviderBuilder `